docs/30DaysOfPWA/advanced-capabilities/07.md

```diff
@@ -4,12 +4,173 @@
+
+It is a best practice to either hide or disable all related UI controls if there is no Push API support in the browser:
+
+```javascript
+if (!('PushManager' in window)) {
+
+    // Code to disable or hide push-related UI controls
+
+    console.log('Push API is not supported');
+    return;
+}
+```
+
+"Subscribe" button handler (for the sake of the example, the code is simplified and doesn't contain error handling):
+
+```javascript
+async function subscribeForPush() {
+    const registration = await navigator.serviceWorker.ready;
+    const vapidPublicKey = "publicKey from Step 1";
+    const pushSubscription = await registration.pushManager.subscribe({
+        userVisibleOnly: true, // Should be always true as currently browsers only support explicitly visible notifications
+        applicationServerKey: urlBase64ToUint8Array(vapidPublicKey) // You will find urlBase64ToUint8Array() function in the link below
+    });
+
+    // Send push subscription to our server to persist it
+    saveSubscription(pushSubscription);
+}
+```
+
+The subscription process is completed. Several possibilities to improve the user experience:
+
+- You can request subscription status from the Messaging Service by using `getSubscription()` to set the initial status for subscription UI controls.
+- Before subscribing, it's a good idea to [check/request permission for the notification using Notifications API](https://docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/notifications-badges#display-notifications-in-the-action-center). As a developer, you will have more flexibility to handle permission requests responses then. If you skip it, calling `subscribe()` will call the notification permission request prompt for you.
+- Don't forget to implement unsubscription logic and place corresponding UI controls. It's always better to let users unsubscribe using your code (they can resubscribe again in your app) than to make them block your origin using browser settings (then to resubscribe they have to unblock your origin in the browser settings again).
+
+You will find an extended code sample [here](https://docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/notifications-badges#step-2---subscribe-to-push-notifications).
+
+3) Time to send a notification! In your backend, you iterate through the push subscription objects you received and saved in Step 2 to send a special HTTPS request to the Messaging Service for each of them. Good to remember, push notifications on the web are not limited to broadcast messages. You can send a specific notification to a specific user device if together with the push subscription object you pass some user-identifying data (like your internal user ID) to the backend, so you can iterate over only specific subscriptions.
+
+To [build](https://datatracker.ietf.org/doc/html/draft-ietf-webpush-protocol) push notification send request, you use:
```

Comment:
Is there any other place we could point people to than this IETF protocol spec? I wouldn't expect people to actually read this to get started with push notifications to be honest.

Reply:
Simplified the statement and moved link to the Resources section